api-v2/src/auth/auth.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { CreateStudentDto } from 'src/student/dto/create-student.dto';
+import { Student } from 'src/student/entities/student.entity';
+import { StudentService } from 'src/student/student.service';
+import { LoginStudentDto } from './dto/login-student.dto';
+import { JwtPayload } from './interfaces/jwt-payload';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    // used to retrieve user from the db based on creds
+    private readonly studentService: StudentService,
+
+    // used to generate and sign a JWT
+    private readonly jwtService: JwtService,
+  ) {}
+
+  /**
+   * Registers a new student in the db and logs the student in.
+   */
+  async register(createStudentDto: CreateStudentDto): Promise<Student> {
+    // create a new student
+    const newStudent = await this.studentService.create(createStudentDto);
+
+    // login the new student by generating a new access token
+    newStudent.accessToken = this._generateAccessToken(newStudent);
+
+    return newStudent;
+  }
+
+  /**
+   * Validates the login creds and logs the student in if the creds are valid.
+   * Else, throws an error.
+   */
+  async login(loginStudentDto: LoginStudentDto): Promise<Student> {
+    const { email, password } = loginStudentDto;
+
+    // find student in db
+    const student = await this.studentService.findByEmail(email);
+
+    if (!student) {
+      throw new Error('Student with given email not found');
+    }
+
+    // validate creds
+    const { password: trueHashedPassword } = student;
+    const isValidPassword = await bcrypt.compare(password, trueHashedPassword);
```

Comment:
ah yep trueHashedPassword is hashed since it's coming from the db(we store hashed passwords in the db using a `@BeforeInsert` in the entity class). Does that answer the question?